### PR TITLE
Offer msg service and facades

### PR DESCRIPTION
**Pull request by JirkaVrbka**

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -9,7 +9,17 @@
...
+    <packaging>jar</packaging>
+
+    <dependencies>
+        <dependency>
+            <groupId>cz.idomatojde</groupId>
+            <artifactId>dao</artifactId>
+            <version>1.0-SNAPSHOT</version>
+            <scope>compile</scope>
+        </dependency>
+    </dependencies>
```
Comment (Michal-MK) [inline api/pom.xml:22]: API should not depend directly on DAO

File: api/src/main/java/cz/idomatojde/dto/user/UserDTO.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
+package cz.idomatojde.dto.user;
+
+import cz.idomatojde.dto.offer.OfferDTO;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * DTO Holding the whole information about {@link cz.idomatojde.entity.User}
```
Comment (Michal-MK) [inline api/src/main/java/cz/idomatojde/dto/user/UserDTO.java]: DAO dependency

File: api/src/main/java/cz/idomatojde/facade/TimetableChatMessageFacade.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+     */
+    void deleteTimetableChatMessage(long id);
+
+    /**
+     * Gets all timetable chat messages belonging to entry with given id
+     *
+     * @param entryId unique identifier for TimetableEntry
+     * @return all timetable chat messages owned by given TimetableEntry
+     */
+    List<TimetableChatMessageDTO> GetAllMessagesOfTimetableEntry(long entryId);
```
Comment (Michal-MK) [inline api/src/main/java/cz/idomatojde/facade/TimetableChatMessageFacade.java]: capitalization

File: business/src/main/java/cz/idomatojde/services/OfferServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    @Override
+    public List<Offer> findByUser(User user) {
+        return offers.findByUser(user);
+    }
+
+    @Override
+    public List<Offer> getActiveOffers() {
+        return offers.getActiveOffers();
+    }
+}
```
Comment (Michal-MK) [inline business/src/main/java/cz/idomatojde/services/OfferServiceImpl.java]: Missing newline as GitHub says

File: dao/src/test/java/cz/idomatojde/TimetableChatMessageTest.java [hunk 5]
```diff
@@ -225,12 +225,46 @@ public void deleteTest() {
...
+        timetableChatMessageDAO.create(anotherChat);
 
         // Act
-        chatMessagesDAO.delete(chat);
+        var messages = timetableChatMessageDAO.getAllMessagesOfUser(offer.getOwner());
 
         // Assert
-        chatMessagesDAO.getById(chat.getId());
+        Assert.assertEquals(messages.size(), 1);
+        Assert.assertEquals(messages.get(0), chat);
```
Comment (Michal-MK) [inline dao/src/test/java/cz/idomatojde/TimetableChatMessageTest.java]: We should really use a single assertion mechanism, and I believe we have more occurences of assertThat() than Assert.assertEquals()

File: api/pom.xml (hunk 1 above)
Comment (JirkaVrbka) [reply api/pom.xml:22]: We need to have this dependency for Category enum. I can create another module for this class but I am not sure if that is the correct solution

File: dao/src/test/java/cz/idomatojde/TimetableChatMessageTest.java (hunk 5 above)
Comment (JirkaVrbka) [reply dao/src/test/java/cz/idomatojde/TimetableChatMessageTest.java]: Ha, missed that. They will be changed

**Pull request merged**